main.tf

```diff
@@ -4,26 +4,30 @@ locals {
 
 # https://www.terraform.io/docs/providers/aws/r/vpn_gateway.html
 resource "aws_vpn_gateway" "default" {
-  count           = local.enabled ? 1 : 0
+  count           = var.vpn_gateway_id != null ? 0 : (local.enabled ? 1 : 0)
   vpc_id          = var.vpc_id
   amazon_side_asn = var.vpn_gateway_amazon_side_asn
   tags            = module.this.tags
 }
 
 # https://www.terraform.io/docs/providers/aws/r/customer_gateway.html
 resource "aws_customer_gateway" "default" {
-  count      = local.enabled ? 1 : 0
+  count      = local.enabled && var.customer_gateway_id == null ? 1 : 0
   bgp_asn    = var.customer_gateway_bgp_asn
   ip_address = var.customer_gateway_ip_address
   type       = "ipsec.1"
-  tags       = module.this.tags
+  tags = merge(
+    module.this.tags,
+    var.customer_gateway_name_override != null ? { Name = var.customer_gateway_name_override } : {},
+    { ip = var.customer_gateway_ip_address }
+  )
```

Comment:
We typically like to avoid supporting name overrides (it happens for sure, but I like to avoid it) because even if you're not following the `module.this` conventions you can create the name by just passing `name`. That said, I'd consider this without the `ip` tag. You can pass that in via the `tags` variable if you're interested in including that as a specific tag. 

Mind cleaning this up a bit and just consuming the module so that your use-case fits or is there a specific need here that I'm missing?